shared/src/simulation/activities/recurring-event.ts

```diff
@@ -49,18 +52,15 @@ export class RecurringEventActivityState implements SimulationActivityState {
 export const recurringEventActivity: SimulationActivity<RecurringEventActivityState> =
     {
         activityState: RecurringEventActivityState,
-        tick(
-            draftState,
-            simulatedRegion,
-            activityState,
-            _tickInterval,
-            terminate
-        ) {
-            if (draftState.currentTime >= activityState.nextOccurrenceTime) {
-                activityState.nextOccurrenceTime +=
+        tick(draftState, simulatedRegion, activityState) {
+            if (
+                draftState.currentTime >=
+                activityState.lastOccurenceTime +
+                    activityState.recurrenceIntervalTime
+            ) {
+                activityState.lastOccurenceTime +=
                     activityState.recurrenceIntervalTime;
```

Comment:
Rather set `lastOccurenceTime` to `draftState.currentTime` to prevent getting more and more off after some time.